BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Batch_Jobs/PueoCall_Array.sh

```diff
@@ -19,67 +19,55 @@
 #NNT=$7
 #Exp=$8
 
-#source /fs/ess/PAS1960/BiconeEvolutionOSC/new_root/new_root_setup.sh
-
-source /fs/ess/PAS1960/buildingPueoSim/set_env.sh
-
-cd $PSIMDIR
-cd outputs/${gen}_outputs
-mkdir -m775 $SLURM_ARRAY_TASK_ID
-cd ../..
-# Need to change IceMC to read in the correct gain files for each run 
-
-
-#this is the command in the XF script although I don't know if we can pass in variables from that script
-#into this one like i and WorkingDir
-#if in the job call we have 
+threads=40
 
 num=$(($((${SLURM_ARRAY_TASK_ID}-1))/${Seeds}+1))
 seed=$(($((${SLURM_ARRAY_TASK_ID}-1))%${Seeds}+1))
 
-run_num=$(((NPOP * gen + num) * 1000 + seed))
+# run number is the unique identifier for each pueoSim Process
+# 40 processes are run per generation per seed
+run_num=$(((NPOP * gen + num) * 1000 + seed * 40))
```

Comment:
Are we using 40 because that's the number of threads? Can that be a variable, or is it covered by using $threads down below in the actually PUEOsim call?

Reply:
40 is the maximum number of threads, so any number of threads will be able to have a unique run_num for each process of pueoSim if we offset each seed by 40. I could change it to be equal to threads, though. They're both the same as long as it's greater than or equal to threads.

Reply:
40 is the maximum number of threads, so any number of threads will be able to have a unique run_num for each process of pueoSim if we offset each seed by 40. I could change it to be equal to threads, though. They're both the same as long as it's greater than or equal to threads.